R/qsys_sge.r

```diff
@@ -13,14 +13,20 @@ SGE = R6::R6Class("SGE",
 
         submit_jobs = function(...) {
             opts = private$fill_options(...)
-            private$job_id = opts$job_name
+            private$job_name = opts$job_name
             filled = private$fill_template(opts)
 
-            success = system("qsub", input=filled, ignore.stdout=TRUE)
-            if (success != 0) {
+            private$qsub_stdout  = system2("qsub", input=filled, stdout = T)
+            
+            status = attr(private$qsub_stdout, "status")
+            success = (!length(status)) || (status != 0)
```

Comment:
`status` can either be `NULL` or `stdout` according to the docs. Please check explicitly for `NULL`, i.e. `is.null(status) || status != 0` (this is just my personal preference here)

Reply:
I do believe my implementation achieves this null check, but I understand why expressing it as I have isn't to your taste. Will do

Reply:
Yes it does, and there's nothing wrong with your check. It's just that I use `is.null` everywhere else in the package, so I think this is less confusing.

Reply:
I also do notice that my status check is kind of weird, it checks empty status, OR non zero value. 

I think that's a logical error, so it probably warrants fixing.

```r
# ==============
# This is fine
status <- NULL
(!length(status) || (status != 0))
#> TRUE

# ==============
# These are wrong
status <- 1
(!length(status) || (status != 0))
#> TRUE

status <- 0
(!length(status) || (status != 0))
#> FALSE

# ========================
# This is the actual behaviour we want
status <- 0
(is.null(status) || (status == 0))
#> TRUE

status <- NULL
(is.null(status) || (status == 0))
#> TRUE


status <- 1
(is.null(status) || (status == 0))
#> FALSE
```

I'll fix this